test/end-to-end/getRedirectionURL.test.js

```diff
@@ -377,5 +375,152 @@ describe("getRedirectionURL Tests", function () {
                 assert.equal(newUserCheck, "thirdpartypasswordless-true");
             });
         });
+
+        describe("No Redirection", function () {
+            describe("Email Password Recipe", function () {
+                let browser;
+                let page;
+
+                before(async function () {
+                    await backendBeforeEach();
+
+                    await fetch(`${TEST_SERVER_BASE_URL}/startst`, {
+                        method: "POST",
+                    }).catch(console.error);
+
+                    browser = await puppeteer.launch({
+                        args: ["--no-sandbox", "--disable-setuid-sandbox"],
+                        headless: true,
+                    });
+
+                    page = await browser.newPage();
+                    // We need to set the localStorage value before the page loads to ensure ST initialises with the correct value
+                    await page.evaluateOnNewDocument(() => {
+                        localStorage.setItem("disableRedirectionAfterSuccessfulSignInUp", "true");
+                        localStorage.removeItem("isNewUserCheck");
```

Comment:
why is this required? What does the `isNewUserCheck` do?

Reply:
This checks that a new user was created (sign up successful) and then no redirection happened. The before and after url would be same in case of a failed sign up as well.